engine/src/main/java/net/jqwik/engine/SourceOfRandomness.java

```diff
@@ -65,7 +65,16 @@ private XORShiftRandom(long seed) {
 			if (seed == 0l) {
 				throw new IllegalArgumentException("0L is not an allowed seed value");
 			}
-			this.seed = seed;
+			this.seed = mix64(seed);
+		}
+
+		/**
+		* See <a href="http://zimbry.blogspot.com/2011/09/better-bit-mixing-improving-on.html">Better Bit Mixing - Improving on MurmurHash3's 64-bit Finalizer</a>
+		*/
+		private static long mix64(long z) {
+			z = (z ^ (z >>> 30)) * 0xbf58476d1ce4e5b9L;
+			z = (z ^ (z >>> 27)) * 0x94d049bb133111ebL;
+			return z ^ (z >>> 31);
```

Comment:
Any reason for choosing this exact mode of scrambling and not just adding 1? Just being curious...

Reply:
Frankly speaking, Copilot suggested me that function for scrambling the value.

I put `this.seed = `, and it autocomplete mix64 literally, except the reference link.
Then I searched the constants online, and I found the corresponding article, so I believe the constants are both good, and ok license-wise.

Reply:
One more thing: it might be that the value will be zero after `mix64`, so it looks like we need to do something about "0L is not an allowed seed value".

What if we replace the result of `mix64` with `0xbf58476d1ce4e5b9L` if it turns out to be `0`?
Then it would have a non-zero starting value, and it would avoid unexpected "all zeroes caused by 0 after mix64"

Reply:
Is the probability higher than 1/2^64? Not sure if it's worth the performance hit and the extra complexity.

Reply:
I believe the performance hit is virtually zero, and the complexity is not there as well: `0` check was there anyway, so I replace the exception with a reasonable value to start with.